homeassistant/components/cast/media_player.py

```diff
@@ -502,6 +511,60 @@ def media_seek(self, position):
         media_controller = self._media_controller()
         media_controller.seek(position)
 
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Implement the websocket media browsing helper."""
+        media: Media = await self.hass.components.media_source.async_find_media(
+            media_content_id
+        )
+        if not media:
+            raise BrowseError("Could not find media for given path.")
+
+        def media_to_dict(media: Media):
```

Comment:
I think that we should have the media source completely handle this. This method should be

```python
return self.hass.components.media_source.async_browse_media(media_content_id)
```

Reply:
Otherwise this method will end up in every integration and all the logic should live in media_source.